tests/common/mod.rs

```diff
@@ -4,30 +4,91 @@
 
 #![cfg(test)]
 
-mod factory_dependency;
-mod ir_artifacts;
-mod libraries;
-mod messages;
-mod optimizer;
-mod remappings;
-mod standard_json;
-mod unsupported_instructions;
-
+mod downloader;
+
+use assert_cmd::Command;
+use era_compiler_solidity::message_type::MessageType;
+use era_compiler_solidity::project::Project;
+use era_compiler_solidity::solc::pipeline::Pipeline as SolcPipeline;
+use era_compiler_solidity::solc::standard_json::input::settings::optimizer::Optimizer as SolcStandardJsonInputSettingsOptimizer;
+use era_compiler_solidity::solc::standard_json::input::settings::selection::Selection as SolcStandardJsonInputSettingsSelection;
+use era_compiler_solidity::solc::standard_json::input::source::Source as SolcStandardJsonInputSource;
+use era_compiler_solidity::solc::standard_json::input::Input as SolcStandardJsonInput;
+use era_compiler_solidity::solc::standard_json::output::error::collectable::Collectable as CollectableError;
+use era_compiler_solidity::solc::standard_json::output::Output as SolcStandardJsonOutput;
+use era_compiler_solidity::solc::Compiler as SolcCompiler;
 use std::collections::BTreeMap;
 use std::collections::BTreeSet;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::str::FromStr;
+use std::sync::Once;
+use std::time::Duration;
+
+/// Synchronization for solc downloads.
+#[cfg(target_family = "unix")]
```

Comment:
What do we do on Windows?

Reply:
Today I tried to make it work for Windows, the main thing that is missing for it it's an older version of `solc` that is downloaded from Solidity mirror, and they are in `zip` archives with additional `dll` libraries... So, for now, on Windows, I'm just using my action that downloads `solc` and unpacks everything.

I'll try to add a proper Windows support in `era-compiler-common`.